sonic-chassisd/scripts/chassisd

```diff
@@ -251,9 +264,33 @@ class ModuleUpdater(logger.Logger):
                                                   (CHASSIS_MODULE_INFO_SERIAL_FIELD, module_info_dict[CHASSIS_MODULE_INFO_SERIAL_FIELD])])
                 self.module_table.set(key, fvs)
 
+                # Construct key for down_modules dict. Example down_modules key format: LINE-CARD0|<hostname>
+                fvs = self.hostname_table.get(key)
+                if isinstance(fvs, list) and fvs[0] is True:
+                    fvs = dict(fvs[-1])
+                    hostname = fvs[CHASSIS_MODULE_INFO_HOSTNAME_FIELD]
+                    down_module_key = key+'|'+hostname
+                else:
+                    down_module_key = None
+
                 if module_info_dict[CHASSIS_MODULE_INFO_OPERSTATUS_FIELD] != str(ModuleBase.MODULE_STATUS_ONLINE):
                     notOnlineModules.append(key)
+                    # Record the time when the module down was detected to track the
+                    # module down time. Used for chassis db cleanup for all asics of the module if the module is down for a 
+                    # long time like 30 mins.
+                    # If the down_module_key is None, hostname is not avaiable for the linecard. i.e., The module was not 
+                    # up before. So No need to do cleanup when hostname is not available at the time module detected down
+                    self.log_warning("Module {} went off-line!".format(key))
+                    if (down_module_key is not None) and (down_module_key not in self.down_modules):
+                        self.down_modules[down_module_key] = {}
+                        self.down_modules[down_module_key]['down_time'] = time.time()
+                        self.down_modules[down_module_key]['cleaned'] = False
```

Comment:
It's discussed in chassis subgroup to add syslog for module state change

Reply:
Added syslog for module state change.